source/index.js

```diff
@@ -56,6 +57,32 @@ module.exports = async (input = 'patch', options) => {
 	const hasLockFile = fs.existsSync(path.resolve(rootDir, 'package-lock.json')) || fs.existsSync(path.resolve(rootDir, 'npm-shrinkwrap.json'));
 	const isOnGitHub = options.repoUrl && hostedGitInfo.fromUrl(options.repoUrl).type === 'github';
 
+	let published = false;
+
+	const rollback = onetime(async () => {
+		console.log('\nPublish failed. Rolling back to the previous state...');
+
+		const lastCommit = await execa.stdout('git', ['log', '-1', '--pretty=%B']);
+		try {
+			if (lastCommit === util.readPkg().version && lastCommit !== pkg.version) {
+				await execa('git', ['tag', '-d', options.version]);
+				await execa('git', ['reset', '--hard', 'HEAD~1']);
+			}
+
+			console.log('Successfully rolled back the project to its previous state.');
+		} catch (error) {
+			console.log(`Couldn't rollback because of the following error:\n${error}`);
+		}
+	});
+
+	process.on('SIGINT', async () => {
```

Comment:
I'm not sure if this works since I haven't had the chance to test it yet.

Reply:
I didn't manage to trigger it.